src/uu/chgrp/src/chgrp.rs

```diff
@@ -21,16 +21,22 @@ use std::os::unix::fs::MetadataExt;
 static ABOUT: &str = help_about!("chgrp.md");
 const USAGE: &str = help_usage!("chgrp.md");
 
-fn parse_gid_and_uid(matches: &ArgMatches) -> UResult<(Option<u32>, Option<u32>, IfFrom)> {
+fn parse_gid_and_uid(matches: &ArgMatches) -> UResult<(Option<u32>, Option<u32>, String, IfFrom)> {
```

Comment:
Maybe it makes sense to make a struct for the `(Option<u32>, Option<u32>, String, IfFrom)`? That would make it a bit easier to read (and make the types shorter).

Reply:
Also the `String` might need to be an `OsString` if it can contain invalid utf-8.

Reply:
The `raw_owner: String` is constructed by either:
 - ` matches.get_one::<String>( ... )` or
 - format of  `entries::uid2usr(uid).unwrap_or_else(|_| uid.to_string())`

I don't know if these two functions are robust enough to invalid utf-8.
If not, it might imply to refactor at least `uucore/*/perms.rs`, `uu/*/chown.rs`, `uu/*/chgrp.rs`. Then it should be useful to open a dedicated issue for it.

Reply:
I refactored `(Option<u32>, Option<u32>, String, IfFrom)` to a struct `GidUidOwnerFilter`.